cookiecutter/zipfile.py

```diff
@@ -0,0 +1,88 @@
+from __future__ import absolute_import
+
+import os
+import requests
+import sys
+from zipfile import ZipFile
+
+from .prompt import read_user_yes_no
+from .utils import make_sure_path_exists, rmtree
+
+
+def prompt_and_delete(path, no_input=False):
+    """Ask the user whether it's okay to delete the previously-downloaded
+    file/directory.
+
+    If yes, deletes it. Otherwise, Cookiecutter exits.
+
+    :param path: Previously downloaded zipfile.
+    :param no_input: Suppress prompt to delete repo and just delete it.
+    """
+    # Suppress prompt if called via API
+    if no_input:
+        ok_to_delete = True
+    else:
+        question = (
+            "You've downloaded {} before. "
+            "Is it okay to delete and re-download it?"
+        ).format(path)
+
+        ok_to_delete = read_user_yes_no(question, 'yes')
+
+    if ok_to_delete:
+        if os.path.isdir(path):
```

Comment:
Zip archives are; but the unpacked zip directory isn't. There's a second usage of this function on line 81 to purge an unpacked zip directory.